aws/resource_aws_ec2_capacity_reservation.go

```diff
@@ -0,0 +1,278 @@
+					ec2.EndDateTypeLimited,
+				}, false),
+			},
+			"ephemeral_storage": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  false,
+			},
+			"instance_count": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"instance_match_criteria": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  ec2.InstanceMatchCriteriaOpen,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.InstanceMatchCriteriaOpen,
+					ec2.InstanceMatchCriteriaTargeted,
+				}, false),
+			},
+			"instance_platform": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.CapacityReservationInstancePlatformLinuxUnix,
+					ec2.CapacityReservationInstancePlatformRedHatEnterpriseLinux,
+					ec2.CapacityReservationInstancePlatformSuselinux,
+					ec2.CapacityReservationInstancePlatformWindows,
+					ec2.CapacityReservationInstancePlatformWindowswithSqlserver,
+					ec2.CapacityReservationInstancePlatformWindowswithSqlserverEnterprise,
+					ec2.CapacityReservationInstancePlatformWindowswithSqlserverStandard,
+					ec2.CapacityReservationInstancePlatformWindowswithSqlserverWeb,
+				}, false),
+			},
+			"instance_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+			"tenancy": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  ec2.CapacityReservationTenancyDefault,
+				ValidateFunc: validation.StringInSlice([]string{
+					ec2.CapacityReservationTenancyDefault,
+					ec2.CapacityReservationTenancyDedicated,
+				}, false),
+			},
+		},
+	}
+}
+
+func resourceAwsEc2CapacityReservationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	opts := &ec2.CreateCapacityReservationInput{
+		AvailabilityZone: aws.String(d.Get("availability_zone").(string)),
+		InstanceCount:    aws.Int64(int64(d.Get("instance_count").(int))),
+		InstancePlatform: aws.String(d.Get("instance_platform").(string)),
+		InstanceType:     aws.String(d.Get("instance_type").(string)),
+	}
+
+	if v, ok := d.GetOk("ebs_optimized"); ok {
+		opts.EbsOptimized = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("end_date"); ok {
+		t, err := time.Parse(time.RFC3339, v.(string))
+		if err != nil {
+			return fmt.Errorf("Error parsing capacity reservation end date: %s", err.Error())
+		}
+		opts.EndDate = aws.Time(t)
+	}
+
+	if v, ok := d.GetOk("end_date_type"); ok {
+		opts.EndDateType = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("ephemeral_storage"); ok {
+		opts.EphemeralStorage = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("instance_match_criteria"); ok {
+		opts.InstanceMatchCriteria = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("tenancy"); ok {
+		opts.Tenancy = aws.String(v.(string))
+	}
+
+	restricted := meta.(*AWSClient).IsChinaCloud()
```

Comment:
Were you able to verify this restriction? If not, let's remove the restriction logic here and below for now.